apps/src/templates/teacherDashboard/EditSectionForm.jsx

```diff
@@ -142,24 +143,47 @@ class EditSectionForm extends Component {
 
   // valid event names: 'Section Setup Complete', 'Section Setup Cancelled'.
   recordSectionSetupExitEvent = eventName => {
-    const {section, courseOfferings, isNewSection} = this.props;
-
+    const {
+      section,
+      courseOfferings,
+      isNewSection,
+      initialCourseId,
+      initialUnitId
+    } = this.props;
+    const versionYear =
+      courseOfferings[section.courseOfferingId].course_versions[
+        section.courseVersionId
+      ].key;
     const course = courseOfferings.hasOwnProperty(section.courseOfferingId)
       ? courseOfferings[section.courseOfferingId]
       : null;
     const courseName = course ? course.display_name : null;
     const courseId = course ? course.id : null;
-
     if (isNewSection) {
       analyticsReporter.sendEvent(eventName, {
+        sectionUnitId: section.unitId,
         sectionCurriculumLocalizedName: courseName,
-        sectionCurriculum: courseId,
+        sectionCurriculum: courseId, //this is course Offering id
+        sectionCurriculumVersionYear: versionYear,
         sectionGrade: section.grade,
         sectionLockSelection: section.restrictSection,
         sectionName: section.name,
         sectionPairProgramSelection: section.pairingAllowed
       });
     }
+    if (
+      eventName === COMPLETED_EVENT &&
+      ((section.courseId && section.courseId !== initialCourseId) ||
+        (section.unitId && section.unitId !== initialUnitId))
+    ) {
+      analyticsReporter.sendEvent(CURRICULUM_ASSIGNED, {
+        previousUnitId: initialUnitId,
+        previousCourseId: initialCourseId,
+        newUnitId: section.unitId,
+        newCourseId: section.courseOfferingId,
+        newVersionYear: versionYear
```

Comment:
Do we want to report the previous version year? Or do we not have access to that?

Reply:
Okay we didn't before, but I was able to pipe it through from teacherSections redux! I've updated the pictures and unit tests to reflect it, and tested that it worked on localhost. In the same commit, I also updated the section.courseId to use courseOfferingId so that we're comparing apples to apples in the analytics.